kornia/geometry/epipolar/projection.py

```diff
@@ -107,6 +107,44 @@ def projection_from_KRt(K: torch.Tensor, R: torch.Tensor, t: torch.Tensor) -> to
 
     return K @ Rt
 
+def KRt_from_projection(P: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+    r"""Get the intrinsics, rotation-matrix and the camera-center.
+
+    This function decomposes the Projection matrix into Camera-Matrix, Rotation Matrix and Translation vector.
+
+
+    Args:
+        P (torch.Tensor): the projection matrix with shape :math:`(B, 3, 4)`.
+
+    Returns:
+        torch.Tensor: The Camera matrix with shape :math:`(B, 3, 3)`.
+        torch.Tensor: The Rotation matrix with shape :math:`(B, 3, 3)`.
+        torch.Tensor: the Translation vector with shape :math:`(B, 3)`.
+
+    """
+
+    assert P.shape[-2:] == (3, 4), "P must be of shape [B, 3, 4]"
```

Comment:
if we want to accept only Bx3x4 I would add an extra assertion to check `len(P.shape) == 3`.

However, could you evaluate if we could easily support `(*, 3, 4)` ? 
[`projection_from_KRt`](https://kornia.readthedocs.io/en/latest/geometry.epipolar.html#kornia.geometry.epipolar.projection_from_KRt) also accepts `Bx3x4` but now looking at the code an I believe that could be easily adapted for `(*,3,4)`. What do you think about it ?

Reply:
I'll definitely verify if my code `KRt_from_projection` can be extended to support `(*, 3, 4)` with minimal changes.
And yes, I too believe the `projection_from_KRt` can be adapted for `(*, 3, 3)`.
Anyways, I shall verify both.
Thanks for the suggestion.

Reply:
sure, let us know whether you want to include this feature in this PR or after merging this one

Reply:
I was thinking of implementing this extension from `B` to `*` after emerging this one.
Currently, I'll add the assertion for `len(P.shape) == 3`